tests/unittests/dataset_tests.py

```diff
@@ -711,6 +711,233 @@ def test_index_in_progress(self):
+            [
+                {
+                    fo.DynamicEmbeddedDocument(label="foo", min=0.1, max=0.9),
+                    fo.DynamicEmbeddedDocument(label="bar", min=0.5, max=0.5),
+                },
+                None,
+            ],
+        )
+
+        summary_fields = dataset.list_summary_fields()
+
+        self.assertTrue(dataset.get_field("gt_label").read_only)
+        self.assertTrue(dataset.get_field("gt_confidence").read_only)
+        self.assertFalse(dataset.get_field("gt_label_counts").read_only)
+        self.assertFalse(dataset.get_field("gt_confidence_by_label").read_only)
+
+        self.assertTrue(dataset.get_field("frames_gt_label").read_only)
+        self.assertTrue(dataset.get_field("frames_gt_confidence").read_only)
+        self.assertFalse(dataset.get_field("frames_gt_label_counts").read_only)
+        self.assertFalse(
+            dataset.get_field("frames_gt_confidence_by_label").read_only
+        )
+
+        self.assertSetEqual(
+            set(summary_fields),
+            {
+                "gt_label",
+                "gt_confidence",
+                "gt_label_counts",
+                "gt_confidence_by_label",
+                "frames_gt_label",
+                "frames_gt_confidence",
+                "frames_gt_label_counts",
+                "frames_gt_confidence_by_label",
+            },
+        )
+
+        db_indexes = dataset.list_indexes()
+
+        self.assertTrue("gt_label" in db_indexes)
+        self.assertTrue("gt_confidence.min" in db_indexes)
+        self.assertTrue("gt_confidence.max" in db_indexes)
+        self.assertFalse("gt_label_counts.label" in db_indexes)
+        self.assertFalse("gt_label_counts.count" in db_indexes)
+        self.assertFalse("gt_confidence_by_label.label" in db_indexes)
+        self.assertFalse("gt_confidence_by_label.min" in db_indexes)
+        self.assertFalse("gt_confidence_by_label.max" in db_indexes)
+
+        self.assertTrue("frames_gt_label" in db_indexes)
+        self.assertTrue("frames_gt_confidence.min" in db_indexes)
+        self.assertTrue("frames_gt_confidence.max" in db_indexes)
+        self.assertFalse("frames_gt_label_counts.label" in db_indexes)
+        self.assertFalse("frames_gt_label_counts.count" in db_indexes)
+        self.assertFalse("frames_gt_confidence_by_label.label" in db_indexes)
+        self.assertFalse("frames_gt_confidence_by_label.min" in db_indexes)
+        self.assertFalse("frames_gt_confidence_by_label.max" in db_indexes)
+
+        update_fields = dataset.check_summary_fields()
+
+        self.assertListEqual(update_fields, [])
+
+        label_upper = F("label").upper()
+        dataset.set_field("gt.detections.label", label_upper).save()
+        dataset.set_field("frames.gt.detections.label", label_upper).save()
+
+        update_fields = dataset.check_summary_fields()
+
+        self.assertSetEqual(
+            set(update_fields),
+            {
+                "gt_label",
```

Comment:
Another noob question, if the field names are upper-cased in line 878 above, not sure why these field names are still in lower case

Reply:
the upper-casing is applied to the values in the field, not the names of the summary fields

Reply:
If that is the case, not sure why `gt_confidence`, `frames_gt_confidence`, `gt_label_counts` and `frames_gt_label_counts` should be in the update_fields. Should these fields be numeric?

Reply:
That's where the note in `check_summary_fields()` documentation comes in; unfortunately it's only a heuristic:

```
        Summary fields may need to be updated whenever there have been
        modifications to the dataset's samples since the summaries were last
        generated.

        Note that inclusion in this list is only a heuristic, as any sample
        modifications may not have affected the summary's source field.
```